README.md

```diff
@@ -65,3 +65,32 @@ For non-standard Linux system like Android [termux](https://termux.com/), use fo
 ```
 make build-termux
 ```
+
+
+# How build tags work.
+
+We use build tags or build constraints to separate build process for different platforms. Build tags are found at the top of the file as a comment. The file will be included only if the tag is present in the build command.
+eg: 
+```go
+//go:build cli
+```
+#### build command using wasm tag
+```sh
+GOOS=js GOARCH=wasm go build -tags=wasm  -o static/main.wasm
+```
+#### build command using cli tag
+
+```sh
+go build -tags=cli  -o static/main.wasm
+```
+
+Note : The first line of the file should be followed by an empty line to make it a valid build tag statement.
+
+If there is an `!` mark then it means the `not` operation on the build tags.
+
+```go
+//go:build !windows
+```
+This will exclude the file if a `windows` build tag is used.
```

Comment:
A bit confused. The compiler will exclude the file if a windows is used? Or it is negating the platform itself (like exclude on all platforms *except* windows?)